### ISSUE TITLE
Could not parse type ListItem

### ISSUE THREAD

**Issue by cecton** [apollographql/apollo-rs#131]
I'm trying to parse this: ~~`query($foo:[Int]){}`~~ `query($foo:[Int]){x}` but I'm getting these errors:

```
ERROR@11:12 "expected a Type" [,
ERROR@11:12 "expected a Variable Definition" [
ERROR@11:12 "expected R_PAREN, got [" [
```

[edit: the example query was wrong because there was no selection. I added a selection. The error did not change.]

**Comment by Danue1** [comment]
I also get an error in graphql-js.
Can you explain why it shouldn't be an error?

https://astexplorer.net/#/gist/f4b869d2b7a7aafeeb0e9a11192a5632/a56c076474eb50e45ed6c49e0d1a32c55b800419

**Comment by Danue1** [comment]
If it is implemented according to the [GraphQL specification](https://spec.graphql.org/October2021), it is correct to throw an error.

The error example `query($foo:[Int]){}` corresponds to [OperationDefinition](https://spec.graphql.org/October2021/#sec-Language.Operations), and the important part is [SelectionSet](https://spec.graphql.org/October2021/#SelectionSet).

```
SelectionSet:
   { Selection <sub>list</sub> }
```
As you can see, there must be at least one Selection.

**Comment by lrlna** [comment]
Thank you for reporting this @cecton! Your example find two(!) errors in our parser:

1. We are not correctly setting an error for an empty selection set.
2. The specific error you're getting is because the parser wasn't checking for LIST_TYPE, and it should!

I'll get you a fix to both of these in a few minutes!

**Comment by Danue1** [comment]
@lrlna 
I am suddenly curious, is there any possibility or plan for apollo-rs to support GraphQL Language Server in the future?
I suddenly wondered when I saw that I started to implement it from Parser and Syntax Analyzer.
(Sorry for the question that is not related to the original issue.)

**Comment by cecton** [comment]
Thank you @lrlna :pray: my issue was not the 1st point but the 2nd actually. I will update this issue for clarity sake.

**Comment by lrlna** [comment]
@cecton fixed it in #135 for you! Do you want me to cut you a release, or are y'all still using commit id?

**Comment by cecton** [comment]
@lrlna I'm fine. We're still using commit id until we publish ourselves. Thanks!
